x-pack/plugins/maps/public/selectors/map_selectors.ts

```diff
@@ -66,9 +70,9 @@ export function createLayerInstance(
   const source: ISource = createSourceInstance(layerDescriptor.sourceDescriptor, inspectorAdapters);
 
   switch (layerDescriptor.type) {
-    case TileLayer.type:
+    case LAYER_TYPE.TILE:
```

Comment:
much more readable, thx.

I think by removing we can also remove all the static `type` properties from the classes. 

e.g. https://github.com/elastic/kibana/blob/7170a64fecb5854ffde180fa98ae1ab72be773dc/x-pack/plugins/maps/public/classes/layers/vector_layer/vector_layer.tsx#L129

We removed these from sources as well, they are not really necessary.